From e9a975f87e14e14f2da96017ee4960db28a50a2a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 24 Apr 2015 19:41:42 +0100
Subject: [PATCH] Escape strings for markup

---
 package_view.py | 8 ++++++--
 widgets.py      | 6 +++++-
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/package_view.py b/package_view.py
index 8e0dcb2..2484da0 100644
--- a/package_view.py
+++ b/package_view.py
@@ -40,6 +40,10 @@ class PackageView(Gtk.VBox):
                           (str,object,object))
     }
 
+    def make_valid(self, istr):
+        ret = istr.replace("&", "&amp;").replace("\"", "&quot;")
+        return ret
+
     def __init__(self, packagedb, installdb):
         Gtk.VBox.__init__(self)
 
@@ -183,7 +187,7 @@ class PackageView(Gtk.VBox):
             self.operation_type = 'INSTALL'
             GObject.idle_add(self.calculate_dependencies, package)
 
-        self.desc.set_markup('<span font=\'30.5\'>“</span><i>  %s  </i><span font=\'30.5\'>”</span>' % str(package.summary))
+        self.desc.set_markup('<span font=\'30.5\'>“</span><i>  %s  </i><span font=\'30.5\'>”</span>' % self.make_valid(str(package.summary)))
 
         if package.source.homepage is not None:
             self.link.set_uri(package.source.homepage)
@@ -196,4 +200,4 @@ class PackageView(Gtk.VBox):
         else:
             self.image.set_from_icon_name(GENERIC, Gtk.IconSize.DIALOG)
             
-        self.description.set_markup(str(package.description))
+        self.description.set_markup(self.make_valid(str(package.description)))
diff --git a/widgets.py b/widgets.py
index 5de3159..6f8fb50 100644
--- a/widgets.py
+++ b/widgets.py
@@ -46,6 +46,10 @@ class PackageLabel(Gtk.VBox):
                           (str,object,object))
     }
 
+    def make_valid(self, istr):
+        ret = istr.replace("&", "&amp;").replace("\"", "&quot;")
+        return ret
+
     def __init__(self, pkg, old_pkg, interactive=False):
         Gtk.VBox.__init__(self)
 
@@ -58,7 +62,7 @@ class PackageLabel(Gtk.VBox):
             self.image.set_from_icon_name(GENERIC, Gtk.IconSize.DIALOG)
 
         self.header.pack_start(self.image, False, False, 5)
-        self.label_title = Gtk.Label("<b>%s</b> - <small>%s</small>\n%s" % (pkg.name, pkg.version, str(pkg.summary)))
+        self.label_title = Gtk.Label("<b>%s</b> - <small>%s</small>\n%s" % (pkg.name, pkg.version, self.make_valid(str(pkg.summary))))
         self.label_title.set_use_markup(True)
         do_justif(self.label_title)
         self.label_title.set_line_wrap(True)
-- 
2.3.2

