From 5fb67c6f54ffe630c821854564888d8ef07c80bf Mon Sep 17 00:00:00 2001
From: Stefan Ric <stfric369@gmail.com>
Date: Sun, 19 Apr 2015 03:35:06 +0200
Subject: [PATCH 4/5] Remove revealer

---
 updates.py | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/updates.py b/updates.py
index 7e71dfc..f6bddaa 100644
--- a/updates.py
+++ b/updates.py
@@ -41,12 +41,8 @@ class UpdatesView(Gtk.VBox):
 
         updates = len(pisi.api.list_upgradable())
         if updates > 1:
-            self.revealer = Gtk.Revealer()
-            self.pack_start(self.revealer, False, True, 0)
-            self.revealer.set_reveal_child(True)
-
             self.toolbar = Gtk.Toolbar()
-            self.revealer.add(self.toolbar)
+            self.pack_start(self.toolbar, False, True, 0)
 
             sep = Gtk.SeparatorToolItem()
             sep.set_expand(True)
-- 
2.3.2

